platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/VirtualLeafRecordSerializer.java

```diff
@@ -114,48 +107,37 @@ public VirtualLeafRecord<K, V> deserialize(final ByteBuffer buffer, final long d
         return new VirtualLeafRecord<>(path, hash, key, value);
     }
 
-    /**
-     * Serialize a data item including header to the output stream returning the size of the data written
-     *
-     * @param leafRecord
-     * 		The virtual record data item to serialize
-     * @param outputStream
-     * 		Output stream to write to
-     */
+    /** {@inheritDoc} */
     @Override
-    public int serialize(final VirtualLeafRecord<K, V> leafRecord, final SerializableDataOutputStream outputStream)
-            throws IOException {
+    public int serialize(final VirtualLeafRecord<K, V> leafRecord, final ByteBuffer buffer) throws IOException {
+        final int initialPos = buffer.position();
+        // path
+        buffer.putLong(leafRecord.getPath());
+        // total length, if variable size
+        if (isVariableSize()) {
+            buffer.putInt(0); // will be updated below
```

Comment:
No. Otherwise it would break everything :) This serializer is used to write leaves to pathToHashKeyValue store. It happens during node cache flushes or during compactions. In both cases, a leaf is first fully written, then the location in the file where it was written is returned to the caller, and only after that the caller updates the index. If there is a read call from a different thread before the index is updated, the item will be read from a different file (where the index still points to).